packages/utils/src/lib/filter.ts

```diff
@@ -0,0 +1,24 @@
+export function filterItemRefsBy<T extends { refs: object[] }>(
+  items: T[],
+  refFilterFn?: (item: T['refs'][number]) => boolean,
+) {
+  return filterBy(
+    items
+      .map(item => ({
+        ...item,
+        refs: filterBy(item.refs, refFilterFn),
+      }))
+      // remove item with empty refs
+      .filter(item => item.refs.length),
+  );
+}
+
+export function filterBy<T = unknown>(
+  refs: T[] = [],
```

Comment:
Why make `refs` parameter optional? What would be the point of calling `filterBy()` without any arguments? :raised_eyebrow:

Reply:
filterBy is removed now